docs/JAX_porting_PyTorch_model.md

```diff
@@ -0,0 +1,1607 @@
+                assert nnx_child is not None, (name, nnx_module)
+                self.copy_module(torch_child, nnx_child, skip_modules=skip_modules)
+            # Copy buffers and params of the module itself (not its children)
+            for name, torch_buffer in torch_module.named_buffers():
+                if "." in name:
+                    # This is child's buffer
+                    continue
+                nnx_buffer = getattr(nnx_module, name)
+                assert isinstance(nnx_buffer, nnx.Variable), (name, nnx_buffer, nnx_module)
+
+                assert nnx_buffer.value.shape == torch_buffer.shape, (
+                    name, nnx_buffer.value.shape, torch_buffer.shape
+                )
+                nnx_buffer.value = jnp.asarray(torch_buffer)
+
+            for name, torch_param in torch_module.named_parameters():
+                if "." in name:
+                    # This is child's parameter
+                    continue
+                nnx_param = getattr(nnx_module, name)
+                assert isinstance(nnx_param, nnx.Param), (name, nnx_param, nnx_module)
+
+                assert nnx_param.value.shape == torch_param.data.shape, (
+                    name, nnx_param.value.shape, torch_param.data.shape
+                )
+                nnx_param.value = jnp.asarray(torch_param.data)
+
+
+def test_modules(
+    nnx_module, torch_module, torch_input, atol=1e-3, mode="eval", permute_torch_input=True, device="cuda"
+):
+    assert torch_input.ndim == 4
+    assert mode in ("eval", "train")
+
+    torch_input = torch_input.to(device)
+    torch_module = torch_module.to(device)
+
+    if mode == "eval":
+        torch_module.eval()
+        nnx_module.eval()
+    else:
+        torch_module.train()
+        nnx_module.train()
+
+    with torch.inference_mode(mode=mode=="eval"):
+        torch_output = torch_module(torch_input)
+
+    if permute_torch_input:
+        torch_input = torch_input.permute(0, 2, 3, 1)
+
+    jax_input = jnp.asarray(torch_input, device=jax.devices(device)[0])
+    jax_output = nnx_module(jax_input)
+    assert jax_output.device == jax.devices(device)[0]
+
+    torch_output = torch_output.detach()
+    if permute_torch_input and torch_output.ndim == 4:
+        torch_output = torch_output.permute(0, 2, 3, 1)
+    jax_expected = jnp.asarray(torch_output)
+
+    assert jnp.allclose(jax_output, jax_expected, atol=atol), (
+        jnp.abs(jax_output - jax_expected).max(),
+        jnp.abs(jax_output - jax_expected).mean(),
+    )
+
+
+t2f = Torch2Flax()
+```
+
+Let us now test our JAX modules. We only test the result of the forward pass in the inference mode such that we avoid disrcrepancies related to random layers like Dropout, StochasticDepth, etc.
+By default, we use absolute error tolerence `1e-3` when comparing the JAX output vs expected PyTorch result.
+For larger modules we set the device to CPU for JAX model to execute on in order to reduce the errors between CPU and CUDA.
```

Comment:
(I think mode is meant to be model, but maybe I am wrong?)
```suggestion
Let us now test our JAX modules. We only test the result of the forward pass in the inference model such that we avoid discrepancies related to random layers like `Dropout`, `StochasticDepth`, etc.
By default, we use absolute error tolerence `1e-3` when comparing the JAX output vs expected PyTorch result.
For larger modules we set the device to CPU for the JAX model to execute on in order to reduce the errors between CPU and CUDA.
```

Reply:
No, inference mode it is a term meaning that we do not track grads on the forward pass and do not update buffers.